### PR TITLE
Add compat data for Window sub features from A to F

### PR DESCRIPTION
**Pull request by keshavmesta**
As discussed in #2011

### CODE REVIEW

**Comment by wbamberg** [comment]
Thanks @keshavmesta !

It looks to me as if this is missing some items. It includes data for the following:

```
cancelIdleCallback
clearImmediate
close
convertPointFromNodeToPage
convertPointFromPageToNode
crypto
customElements
devicePixelRatio
dialogArguments
event
frameElement
fullScreen
```

Looking at the sidebar for `Window`, I see these properties:

```
applicationCache
caches*
closed
console
controllers
crypto
customElements
defaultStatus
devicePixelRatio
dialogArguments
directories
document
frameElement
frames
fullScreen
```

...and these methods:

```
alert()
atob()*
back()
blur()
btoa()*
cancelAnimationFrame()
cancelIdleCallback()
captureEvents()
clearImmediate()
clearInterval()*
clearTimeout()*
close()
confirm()
convertPointFromNodeToPage()
convertPointFromPageToNode
createImageBitmap()*
dump()
event
fetch()*
find()
focus()
forward()
```

The ones I put an asterisk `*` by are part of the `WindowOrWorkerGlobalScope` interface, so (as I understand it) they are to be omitted from the `Window` data. But I'm not sure why the rest are missing.

**Comment by keshavmesta** [comment]
@wbamberg yes there is no compat data available for the skipped ones on MDN.

**Comment by wbamberg** [comment]
Thanks for the clarification @keshavmesta  :)

File: api/Window.json [hunk 1]
```diff
@@ -125,6 +125,262 @@
...
+              "version_added": true
+            },
+            "edge": {
+              "version_added": true
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": true
```
Comment (wbamberg) [inline api/Window.json]: I'm not sure why this omitted the note: https://developer.mozilla.org/en-US/docs/Web/API/Window/close#Browser_compatibility.

File: api/Window.json [hunk 2]
```diff
@@ -125,6 +125,262 @@
...
+            "ie": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": null
+            },
+            "opera_android": {
+              "version_added": null
+            },
+            "safari": {
```
Comment (wbamberg) [inline api/Window.json]: This also omits a note saying that the Safari version is prefixed - we should capure this, but using `prefix` not a note.

That note also refers to IE, including a link that now redirects to the MDN page :). We should just scrap that part of the note.

File: api/Window.json [hunk 3]
```diff
@@ -125,6 +125,262 @@
...
+              "version_added": true
```
Comment (wbamberg) [inline api/Window.json]: Same issue as above for `convertPointFromNodeToPage`.

File: api/Window.json [hunk 4]
```diff
@@ -125,6 +125,262 @@
...
+            },
+            "firefox": {
+              "version_added": true
+            },
+            "firefox_android": {
+              "version_added": true
+            },
+            "ie": {
+              "version_added": "11"
+            },
```
Comment (wbamberg) [inline api/Window.json]: Prefixed in IE according to the table in the page.

File: api/Window.json [hunk 5]
```diff
@@ -228,6 +484,267 @@
...
+              "version_added": true
+            },
+            "edge": {
+              "version_added": true
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": true
```
Comment (wbamberg) [inline api/Window.json]: 49 according to the page

File: api/Window.json [hunk 6]
```diff
@@ -228,6 +484,267 @@
...
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": null
+            },
+            "firefox_android": {
+              "version_added": true
```
Comment (wbamberg) [inline api/Window.json]: It seems unlikely that Firefox Android supports this if Desktop doesn't. I'd make this `null`.

File: api/Window.json [hunk 7]
```diff
@@ -228,6 +484,267 @@
...
+          }
+        }
+      },
+      "frameElement": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Window/frameElement",
+          "support": {
+            "chrome": {
+              "version_added": "4",
+              "notes": "?"
```
Comment (wbamberg) [inline api/Window.json]: Let's just omit this. As a note it isn't helpful. If we don't know, use `true` in `version_added`.

File: api/Window.json [hunk 8]
```diff
@@ -228,6 +484,267 @@
...
+              "version_added": "1",
+              "notes": "1.7 or earlier"
+            },
+            "firefox_android": {
+              "version_added": true,
+              "notes": "1.7 or earlier"
+            },
+            "ie": {
+              "version_added": "5.5",
+              "notes": "?"
```
Comment (wbamberg) [inline api/Window.json]: Same comment here, omit this.

File: api/Window.json [hunk 9]
```diff
@@ -228,6 +484,267 @@
...
+            },
+            "opera": {
+              "version_added": "12.1",
+              "notes": "?"
```
Comment (wbamberg) [inline api/Window.json]: Same comment here, omit this.

File: api/Window.json [hunk 10]
```diff
@@ -228,6 +484,267 @@
...
+            },
+            "opera_android": {
+              "version_added": null
+            },
+            "safari": {
+              "version_added": "4",
+              "notes": "?"
```
Comment (wbamberg) [inline api/Window.json]: Same comment here, omit this.

File: api/Window.json [hunk 11]
```diff
@@ -228,6 +484,267 @@
...
+            },
+            "edge": {
+              "version_added": true
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": "1",
+              "notes": "1.7 or earlier"
```
Comment (wbamberg) [inline api/Window.json]: Same comment here, omit this. Use "1" or `true`.

File: api/Window.json [hunk 12]
```diff
@@ -228,6 +484,267 @@
...
+            },
+            "firefox_android": {
+              "version_added": true,
+              "notes": "1.7 or earlier"
```
Comment (wbamberg) [inline api/Window.json]: Again, omit this.

File: api/Window.json [hunk 13]
```diff
@@ -228,6 +484,267 @@
...
+            "samsunginternet_android": {
+              "version_added": null
+            },
+            "webview_android": {
+              "version_added": null
+            }
+          },
+          "status": {
+            "experimental": false,
+            "standard_track": true,
```
Comment (wbamberg) [inline api/Window.json]: This is not standard according to the page "window.fullScreen is not part of any W3C specification or technical recommendation.".

File: api/Window.json [hunk 14]
```diff
@@ -2595,4 +3112,4 @@
       }
     }
   }
-}
+}
```
Comment (wbamberg) [inline api/Window.json]: Files should end with a newline.

**Comment by wbamberg** [comment]
I've pushed a commit that (I hope) addresses the review comments I had for this PR. I don't know if it needs an extra review go-round or not :).

**Comment by keshavmesta** [comment]
Thanks @wbamberg for taking care of those review comments  👍

**Pull request merged**